test/parallel/test-process-chdir.js

```diff
@@ -33,10 +33,9 @@ process.chdir('..');
 assert.strictEqual(process.cwd().normalize(),
                    path.resolve(tmpdir.path).normalize());
 
-const errMessage = /^TypeError: Bad argument\.$/;
-assert.throws(function() { process.chdir({}); },
-              errMessage, 'Bad argument.');
-assert.throws(function() { process.chdir(); },
-              errMessage, 'Bad argument.');
-assert.throws(function() { process.chdir('x', 'y'); },
-              errMessage, 'Bad argument.');
+const err = {
+  code: 'ERR_INVALID_ARG_TYPE',
+  message: /The "directory" argument must be of type string/
+};
+common.expectsError(function() { process.chdir({}); }, err);
+common.expectsError(function() { process.chdir(); }, err);
```

Comment:
Please keep the third test but just change it to: `process.chdir('x', 'y'); // Should not throw`. That makes sure we have no regression in case the function is changed at any point in time.

Besides that it would also be good to keep `assert.throws` as it has a much nicer error output and is otherwise almost identical.

Reply:
> Please keep the third test but just change it to: process.chdir('x', 'y'); // Should not throw. That makes sure we have no regression in case the function is changed at any point in time.

I don't think we need this test. I removed the check because in other functions we never throw when too many arguments are passed.

> Besides that it would also be good to keep assert.throws as it has a much nicer error output and is otherwise almost identical.

I know our opinions differ here, but I really like to be able to pass a RegExp. I don't have to put the exact string, so:
- I don't need to make the test fail to know exactly what to put for the message.
- I can use one object validator for multiple test cases (that's what I did here, as error the message is not exactly the same for those cases).
- Message validator is shorter, so less likely to go over 80 chars and also less like to have to be updated in case we change something in the message format.